pkg/core/label_list.go

```diff
@@ -133,7 +139,15 @@ func listLabelsChan(repo string, stores context2.Stores, prefix string, opts ...
 	keysChan := make(chan keyBatchEvent, 1)
 
 	iterator := func(next string) ([]string, string, error) {
-		return GetLabelStore(stores).KeysPrefix(context.Background(), next, model.GetArchivePathPrefixToLabels(repo, prefix), "", settings.batchSize)
+		if settings.label != "" {
+			return GetLabelStore(stores).KeysPrefix(context.Background(), next,
+				model.GetArchivePathPrefixToLabels(repo, prefix),
+				"", settings.batchSize)
+		} else {
+			versions, versionsErr := GetLabelStore(stores).(storage.StoreVersioned).
```

Comment:
why do we force an error here, since the underlying store may be legit not implementing this interface?
I'd suggest falling back on the previous implementation when `KeyVersions` is not available.

Reply:
i am not sure what you mean by "force an error".  if `versionsErr` is `nil`, then `listLabelsChan` also returns `nil` error.

Reply:
`versionErr!= nil` here means that the type assertion failed. This means that the underlying storage does not support versioning. I am not seeing this as a sufficient cause for returning an error and stopping the process.

Another way of seeing this is to keep the error (as you do), and document with the option (or func arg, see discussion above) that 
_this feature is not supported for non-versioned stores_ (that is localfs and our current implementation of the S3 store)

Reply:
this was changed.

my mistake:  type assertions look like method chaining, but they aren't.